### PR TITLE
Reveal deposits using `typescript` library

### PR DESCRIPTION
**Pull request by lukasz-zimnoch**
Closes: #146 
Depends on: #297 

Here we enhance the `TBTC` interface with the `revealDeposit` function that allows revealing a deposit to the Bridge on-chain contract. This function should nicely play with deposits created through the `makeDeposit` or `createDepositTransaction` functions thus allowing to build a complete deposit flow.

### CODE REVIEW

**Comment by lukasz-zimnoch** [comment]
Ready for review. Will undraft once #297 lands on main

File: typescript/src/chain.ts [hunk 1]
```diff
@@ -31,6 +32,19 @@ export interface Bridge {
     mainUtxo: UnspentTransactionOutput
   ): Promise<void>
 
+  /**
+   * Reveals a given deposit to the on-chain contract.
+   * @param depositTx - Deposit transaction data.
+   * @param depositOutputIndex - Index of the deposit transaction output that
+   *        funds the revealed deposit.
+   * @param deposit - Data of the revealed deposit.
```
Comment (tomaszslabon) [inline typescript/src/chain.ts:40]: `@param` should be without a dot at the end.

Comment (lukasz-zimnoch) [reply typescript/src/chain.ts]: https://github.com/keep-network/tbtc-v2/pull/298/commits/b115e9a2112233ac19ebbc455dbba8d16471797f

File: typescript/src/deposit.ts [hunk 2]
```diff
@@ -285,5 +297,26 @@ export async function createDepositAddress(
...
 }
 
-// TODO: Implementation and documentation.
-export async function revealDeposit(): Promise<void> {}
+/**
+ * Reveals the given deposit to the on-chain Bridge contract.
+ * @param utxo - Deposit UTXO of the revealed deposit.
+ * @param deposit - Data of the revealed deposit.
+ * @param bitcoinClient - Bitcoin client used to interact with the network.
+ * @param bridge - Handle to the Bridge on-chain contract.
```
[the remark this answers is not in the document]
Comment (lukasz-zimnoch) [reply typescript/src/deposit.ts]: https://github.com/keep-network/tbtc-v2/pull/298/commits/b115e9a2112233ac19ebbc455dbba8d16471797f

File: typescript/src/deposit.ts [hunk 3]
```diff
@@ -66,14 +72,14 @@ export interface Deposit {
  * @param bitcoinClient - Bitcoin client used to interact with the network.
  * @param witness - If true, a witness (P2WSH) transaction will be created.
  *        Otherwise, a legacy P2SH transaction will be made.
- * @returns Empty promise.
+ * @returns Deposit UTXO.
```
Comment (tomaszslabon) [inline typescript/src/deposit.ts:75]: Maybe we could add more information on what tis returned deposit really is? Something like:
`@returns The UTXO that will be created by the deposit transaction.`

Comment (lukasz-zimnoch) [reply typescript/src/deposit.ts]: https://github.com/keep-network/tbtc-v2/pull/298/commits/416f2f55013e3a2d3684ee69ef9594a8022e7665

**Review by tomaszslabon — APPROVED** [review]

**Pull request merged**